src/services/database_service.py

```
@@ -0,0 +1,535 @@
+                            successful_inserts += 1
+                        except Exception as individual_error:
+                            logger.warning(f"Failed to insert individual record: {individual_error}")
+
+                    if successful_inserts > 0:
+                        logger.info(
+                            f"Successfully inserted {successful_inserts}/{len(batch_data)} records individually"
+                        )
+
+
+# Global service instance for backward compatibility (lazy initialization)
+_database_service = None
```

Pylint (3.3.7)

[convention] Constant name "_database_service" doesn't conform to UPPER_CASE naming style

(C0103)